### ISSUE TITLE
Parsing in SingleValue mode is not referentially transparent

### ISSUE THREAD

**Issue by chwthewke** [circe/circe-fs2#54]
Hello,

The short version of my issue goes something like

```scala
val stream: Stream[IO, Byte] = ??? // some stream I have
val parseOnce = stream.through(byteParser(AsyncParser.SingleValue)).compile.toVector
parseOnce.unsafeRunSync() // OK
parseOnce.unsafeRunSync() // fails with a parsing failure
```

This happens even when the stream is a plain in-memory one: `Stream.emits(... some bytes ...)`.
I guess we will find ourselves in agreement that this is not only very surprising, but also a RT bug.

Cause : it seems the underlying jawn parser, created in `ParsingPipe.makeParser` is reused (including some internal state?) and in the second run, sees the input stream repeated twice, which fails to parse in single value mode.

I've got a pull request ready to go with:
1) A test evidencing the issue. (For some reason the error reporting with cats' triple equal was somehow broken, so I left it at a universal equals)
2) A possible fix. Although there might be some things to discuss, as it brings in `Sync` as a constraint on the effect type (and I kinda think it _must_ do so).

Thanks for reading

**Comment by travisbrown** [comment]
Thanks for reporting this! I'll take a look at the proposed fix asap.

**Comment by BenFradet** [comment]
fixed in #55 closing
